### DISCUSSION TITLE
Should cnquery warn users about wrong syntax within predicates/assertions?

### DISCUSSION THREAD

**Discussion by mondoo-Alexander** [mondoohq/cnquery#435] [Q&A]
Currently, when I for example mistype a query and end up with the following:
`users.where( uid = '82' )`
...I will get the same result as if I had typed:
`users` (or the older `users.list`)
This happens because I wrote `=` instead of `==`, which would have returned the expected result. In Typescript, it seems to do the exact same as here, whereas it does not work and give the user an error message in C#.

Should the user be warned that their query was faulty or whether we want to keep this behavior as the default?

**Comment by benr** [comment]
Staying consistent with TypeScript has it's advantages.  Using `where()` on a field which doesn't exist (ie: `users.where( foo )`) throws a compilation error and using a non-sensical operator (ie: `+` ) throws a "cannot find operator handler".   Passing too many arguments retuns a "failed to compile: too many arguments when calling 'where', only 1 is supported" error.   

Therefore, I think it would be reasonable to return a `failed to compile: assignment not allowed when calling 'where'` error.

**Reply by benr** [reply]
@arlimus Your opinion?

**Reply by imilchev** [reply]
The equivalent of this MQL statement actually compiles just fine in Typescript and returns the same result as Typescript. You can check it [here](https://www.typescriptlang.org/play?#code/FDDGHsDsGcBcAICGAnZ8C88DaAiROBdECGBAMwEsAbWAU2VoBMMlUA6Sm+gCgA8MAfPH6YcAIxwBKANxgo0cFVpsq4AObdOdBo0kgA9PvgAVcPCUpI8ALbgGSMeACuCWAAta5xJDVPEa2gAaeFAqClAAawdwADdPCiscAFFeRGsAByVoHHg7eBwAdTdEWAByaHgAOVoAdxw2YEN4AHl3ehqKaCD4AIQ4FDpmMQBPeAZbGIS1eDaukPBra1pIWArvZjb4GrsqZk74Yec0TMRhtWRnSEYG+HggA). 

In C#, however, if you do the same type of thing you get a compilation error.

Personally, I prefer the C# way of handling this situation

**Reply by joelddiaz** [reply]
If I'm reading this correctly, the `where` is the "test" to execute against each element before deciding whether it passes the filter. In the same way that an `if (myString = "new value")` would go into the `if` block no mater the previous/current value of `myString`, returning every element doesn't seem surprising (even though it wasn't what was intended) ¯\_(ツ)_/¯
